### ISSUE TITLE
We should provide a better error message when removing the visibility of the field

### ISSUE THREAD

**Issue by dragonsahead** [metabase/metabase#42309]
### Describe the bug

If you create a query with a field that is taken out of visibility later, we send an error saying that the field does not exist, but it does, we just remove the visibility of it. We should send a message like "this question has a hidden field and cannot be processed or something"

### To Reproduce

1) create a question like sum of total group by prod->category
2) save the question
3) now set that field to "do not include" in table metadata
4) run the question again and see the error ("column source.total does not exist")

### Expected behavior

We should send the correct message

### Logs

NA

### Information about your Metabase installation

```JSON
v49
```

### Severity

P3

### Additional context

It has always been like that for sure

**Comment by thebiglabasky** [comment]
This looks related to field references lost in aliases translation which is part of the larger refactoring that's ongoing with introducing stable IDs for fields. Moving the the querying team

**Comment by ranquild** [comment]
Works now in v57 

<img src="https://uploads.linear.app/a24a1a82-ddc8-4567-aacc-c3804ea2385e/e24f651f-a54d-4f76-a80c-233a30019e89/0ccf2c5e-5de7-4cae-bc3c-1f90fc3c23ab?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2EyNGExYTgyLWRkYzgtNDU2Ny1hYWNjLWMzODA0ZWEyMzg1ZS9lMjRmNjUxZi1hNTRkLTRmNzYtYTgwYy0yMzNhMzAwMTllODkvMGNjZjJjNWUtNWRlNy00Y2FlLWJjM2MtMWY5MGZjM2MyM2FiIiwiaWF0IjoxNzY2MDg1NzI1LCJleHAiOjE3OTc2NTYyODV9.GBKjhdxtNhmdyMks80u9nu8nNuLOPmP74WJDvYvXecU " alt="Screenshot 2025-12-18 at 14.21.46.png" width="1728" data-linear-height="733" />

<img src="https://uploads.linear.app/a24a1a82-ddc8-4567-aacc-c3804ea2385e/b94da86e-800f-4472-a95a-39e089caef94/8ff8badd-a889-4f02-8eb2-3ed84f3d1abb?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2EyNGExYTgyLWRkYzgtNDU2Ny1hYWNjLWMzODA0ZWEyMzg1ZS9iOTRkYTg2ZS04MDBmLTQ0NzItYTk1YS0zOWUwODljYWVmOTQvOGZmOGJhZGQtYTg4OS00ZjAyLThlYjItM2VkODRmM2QxYWJiIiwiaWF0IjoxNzY2MDg1NzI1LCJleHAiOjE3OTc2NTYyODV9.mUIdUY21EXo2lqXhUXgXLben2o4j4Atp009jk_1GnKU " alt="Screenshot 2025-12-18 at 14.21.42.png" width="1723" data-linear-height="807" />

**Comment by rileythomp** [comment]
@dragonsahead @ranquild Is the expected behaviour that this query runs but just doesn't show the field in the query builder? You can still see it in the native query preview. Or should we be returning an error if the field is set to do not include?

**Comment by dragonsahead** [comment]
This one is so old that I no longer remember why I created this :S. Product should chime in here definitely

**Comment by technomancy** [comment]
Yeah, given that there's no more error here, I think this is working as it should. If product wants different behavior here, that's fine but it's not a bug.
